rhizome/lib/vm_setup.rb

```diff
@@ -151,7 +150,29 @@ def parse_routes(routes)
     fail "No default route found in #{routes_j.inspect}"
   end
 
-  def cloudinit(unix_user, public_key, private_subnets)
+  def routes4(ip4, ip4_local)
+    return unless ip4
+    vm_sub = NetAddr::IPv4Net.parse(ip4)
+    local_ip = NetAddr::IPv4Net.parse(ip4_local)
+    vm, vetho, vethi = [vm_sub.to_s,
+      local_ip.network.to_s,
+      local_ip.next_sib.network.to_s]
+
+    r "ip addr add #{vetho}/32 dev vetho#{q_vm}"
+    r "ip route add #{vm} dev vetho#{q_vm}"
+    r "echo 1 > /proc/sys/net/ipv4/conf/vetho#{q_vm}/proxy_arp"
+
+    r "ip -n #{q_vm} link set dev lo up"
```

Comment:
Do we use the loopback interface? This affects more than just ipv4 routes, if we need it up, it sort of doesn't belong here.

Reply:
This was there from the manual ip assignment days. While trying many different commands, it sneaked its way into here.